software/firmware/configuration.py

```diff
@@ -213,3 +213,39 @@ def load_config(cls, config_spec: ConfigSpec):
     def delete_config(cls):
         """Deletes the config file, effectively resetting to defaults."""
         delete_file(ConfigFile.config_filename(cls))
+
+
+class ConfigSettings:
+    """Collects the configuration settings into an object with attributes instead of a dict with keys
+
+    Dict keys are converted to upper-case strings with underscores
+    """
+
+    def __init__(self, d):
+        """Constructor
+
+        @param d  The raw dict loaded from the configuration file
+        """
+        self.__dict__ = {}
```

Comment:
[required] This isn't used and can be removed.

Reply:
This is required to allow `setattr` and `getattr` to work; if `self.__dict__` isn't defined those functions will fail.

Reply:
Oh, my bad, It's been a while since I've used those I guess.